components/Auth/Auth.js

```diff
@@ -0,0 +1,184 @@
+import { Component } from "react";
+import PropTypes from "prop-types";
+import axios from "axios";
+import get from "lodash/get";
+import { toast } from "react-toastify";
+
+import AuthContext from "../../contexts/AuthContext";
+
+class Auth extends Component {
+  static defaultProps = {
+    children: null,
+  };
+
+  static propTypes = {
+    children: PropTypes.node,
+  };
+
+  state = {
```

Comment:
make use of constructor to initialise state, I haven't tried it initializing state outside constructor may be this gets initialized multiple times as it is outside the constructor and whatever is outside constructor gets re-rendered multiple times.

Reply:
no this only gets initialized once, there is no need of constructor since I don't have to bind any methods to the component instance